add the function of CI

Rwatana opened the pull request:
Add the CI

Commit:
add the function of CI

Commit:
change1

Commit:
change2

Commit:
change2

Commit:
change2

Commit:
change2

Commit:
change2

Commit:
change2

Commit:
change2

Rwatana commented:
@nayuta-ai 
## Topic
I think I have added an CI function by using Github actions and test the file.
Please check my code.

## Time
2h

nayuta723 commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,22 @@
+name: Go
+
+on:
+  push:
```
Please change the target on push event.
This implementation only supports this branch, but ci should support all pull request.

nayuta723 commented on test/src/resource_test.go:
```diff
@@ -0,0 +1,65 @@
+package main
```
Divide into test file and src file.

nayuta723 commented on test/src/resource_test.go:
```diff
@@ -0,0 +1,65 @@
+package main
```
This file structure is strange.
test directory contains only test file.

nayuta723 replied:
src directory contains some implementation, but containing only test file is strange.

nayuta723 replied:
Refer to https://github.com/okuda-seminar/client-go/pull/2#discussion_r1311178569

Commit:
make the test directory to contain only test file and make the CI work every branch

Rwatana commented:
@nayuta-ai 
## Topic
- I think I changed to work CI in all branches by changing them below. But, I couldn't
confirm them. Does it meet the demands?
```
on:
  push:
    branches: ['*']
  pull_request:
    branches: ['*']
```
- I changed the structure of directory. Please check my change.

nayuta723 commented:
@Rwatana 
Yes, It does!
Done!

nayuta723 commented:
Also, PR3 should only contain CI code. But it contains the test file related to PR2. Because it is strange, so please change.
